lib/api/core/models/repositories/repository.js

```diff
@@ -97,15 +103,19 @@ class Repository {
 
           return this.fromDTO(dto);
         }
+
         return null;
       })
       .catch(error => {
-        if (error.status === 404) {
-          // no content found, we return null without failing
-          return Bluebird.resolve(null);
+        if (error.status !== 404) {
+          return Bluebird.reject(error);
+        }
+
+        if (! options.returnNull) {
+          return Bluebird.reject(new NotFoundError(`Unable to find ${this._objectName()} with id '${id}'`));
         }
 
-        return Bluebird.reject(error);
+        return Bluebird.resolve(null);
```

Comment:
I strongly disagree with this, as this introduces an unneeded inconsistency in the repositories API.
I would prefer that you remove that `returnNull` option and, in the persistXxx functions, that you try...catch the result of the yielded call and act appropriately if an exception status equal to 404 is received.

Reply:
Much clever solution. I wasn't comfortable with the `persistUser` function, I never use this kind of javascript black magic before. I will find out :)